### PR TITLE
fix: [CDS-46264]: added pre commit hook and custom script to block changing formName and data-tooltip-id identifiers

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

Screenshot 2022-11-09 at 5 27 40 PM
Screenshot 2022-11-09 at 5 27 46 PM

Moving Forward we cannot add a variable by the name of formName / data-tooltip-id the lint rule stops it. Declare these things inline where we want to pass.

Screenshot 2022-11-09 at 5 28 24 PM
Screenshot 2022-11-09 at 5 30 23 PM

We cannot change the existing formNames and tooltipid's as these break the reference, If really. required please do a force commit.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -249,7 +249,7 @@
       "sh scripts/license/stamp.sh"
     ],
     "*.{ts,tsx}": [
-      "eslint --rulesdir ./scripts/eslint-rules --resolve-plugins-relative-to",
+      "eslint --rulesdir ./scripts/eslint-rules",
```
Comment (udayvunnam) [inline package.json:252]: please check if this causes any rules to skip

**Review by udayvunnam — DISMISSED** [review]

File: scripts/eslint-rules/no-restricted-variables.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      description: 'Disallow specific variable identifiers'
+    }
+  },
+
+  create: function (context) {
+    return {
+      VariableDeclaration(node) {
+        const variableName = get(node, 'declarations[0].id.name')
+        if (restrictedVariables.includes(variableName)) {
+          return context.report({ node, message: `retricted variable name ${variableName} not allowed` })
```
Comment (AshwinBhatkal-Harness) [inline scripts/eslint-rules/no-restricted-variables.js:24]: typo: restricted

Comment (vikrant-harness) [reply scripts/eslint-rules/no-restricted-variables.js:24]: will correct this

Comment (vikrant-harness) [reply scripts/eslint-rules/no-restricted-variables.js]: Done

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

File: scripts/eslint-rules/no-restricted-variables.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+const { get } = require('lodash')
+
+const restrictedVariables = ['formName', 'data-tooltip-id']
```
Comment (abhinavrastogi-harness) [inline scripts/eslint-rules/no-restricted-variables.js:10]: can we take the list of variable names from eslint config instead? it'll make the plugin more re-usable.

File: scripts/eslint-rules/no-restricted-variables.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+/*
```
Comment (abhinavrastogi-harness) [inline scripts/eslint-rules/no-restricted-variables.js:1]: can we move this to the `jarvis` repo where we keep all resuable tools?
https://github.com/harness/jarvis

Comment (vikrant-harness) [reply scripts/eslint-rules/no-restricted-variables.js:1]: Discussed this with @vkbansal , we will move this lint rule to jarvis eventually.

**Review by vkbansal-harness — APPROVED** [review]

File: scripts/eslint-rules/no-restricted-variables.js (hunk 3 above)
Comment (knagurski) [reply scripts/eslint-rules/no-restricted-variables.js:10]: I'm pretty sure it's impossible to create a variable called `data-tooltip-id` in JS.

**Pull request merged**